crates/fluvio-controlplane-metadata/src/smartmodule/spec.rs

```diff
@@ -109,7 +110,7 @@ pub struct SmartModuleWasmSummary {
 pub struct SmartModuleWasm {
     pub format: SmartModuleWasmFormat,
     #[cfg_attr(feature = "use_serde", serde(with = "base64"))]
-    pub payload: Vec<u8>,
+    pub payload: ByteBuf,
```

Comment:
let's remove `pub` signature, we don't need to expose it

Reply:
The fluvio-sc-schema uses the field:
https://github.com/infinyon/fluvio/blob/6c57ed0d6d72ddd31d1c9cd59910b6b66f4fa79a/crates/fluvio-sc-schema/src/smartmodule/mod.rs#L21

I think we could expose it via method call (similar to the getter pattern)?

Reply:
we could create a `fn payload_len`

Reply:
Its good for this use case, but we also consume the `payload` directly for the `SmartModuleInvocationWasm` enum here:

```
error[E0616]: field `payload` of struct `SmartModuleWasm` is private
   --> crates/fluvio-spu/src/smartengine/context.rs:253:78
    |
253 |                 wasm: SmartModuleInvocationWasm::AdHoc(smartmodule.spec.wasm.payload.into()),
```

I think we could think of an API for these use case on a next PR? In order to encapsulate these fields, and
also provide some unit tests to these.